enterprise/cli/licenses.go

```diff
@@ -136,6 +137,71 @@ func validJWT(s string) error {
 }
 
 func (r *RootCmd) licensesList() *clibase.Cmd {
+	type tableLicense struct {
+		ID         int32     `table:"id,default_sort"`
+		UUID       uuid.UUID `table:"uuid" format:"uuid"`
+		UploadedAt time.Time `table:"uploaded_at" format:"date-time"`
+		// Features is the formatted string for the license claims.
+		// Used for the table view.
+		Features  string    `table:"features"`
+		ExpiresAt time.Time `table:"expires_at" format:"date-time"`
+		Trial     bool      `table:"trial"`
+	}
+
+	formatter := cliui.NewOutputFormatter(
+		cliui.ChangeFormatterData(
+			cliui.TableFormat([]tableLicense{}, []string{"UUID", "Expires At", "Uploaded At", "Features"}),
+			func(data any) (any, error) {
+				list, ok := data.([]codersdk.License)
+				if !ok {
+					return nil, xerrors.Errorf("invalid data type %T", data)
+				}
+				out := make([]tableLicense, 0, len(list))
+				for _, lic := range list {
+					var formattedFeatures string
+					features, err := lic.FeaturesClaims()
+					if err != nil {
+						formattedFeatures = xerrors.Errorf("invalid license: %w", err).Error()
+					} else {
+						var strs []string
+						if lic.AllFeaturesClaim() {
+							strs = append(strs, "all")
```

Comment:
Do we need to add all zero features if we add all?

```
UUID                                  UPLOADED AT           FEATURES                                                                           EXPIRES AT
1d6fab30-b9fe-40f1-8510-f054bbe252ab  2022-11-05T16:45:44Z  audit_log=1, user_limit=5                                                          2023-08-26T22:19:20Z
6a6fa845-b022-4980-953f-982617bdf74e  2023-01-19T22:01:43Z  all, workspace_quota=0, audit_log=0, browser_only=0, rbac=0, scim=0, user_limit=0  2023-10-10T21:01:34Z
```

In this case, I think simply omitting those would make sense: `all, workspace_quota=0, audit_log=0, browser_only=0, rbac=0, scim=0, user_limit=0` -> `all`.

Reply:
I was honestly curious why we even include the claim if it's 0.

You are right though, I'll omit 0s.